orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -1183,89 +1187,89 @@ def run_step(model, input_left, input_right):
-    rhs_candidates = []
-    for comb in combs:
-        rhs_candidates += list(itertools.permutations(comb))
-
-    all_cases = []
-    for lhs_candidate in lhs_candidates:
-        for rhs_candidate in [list(candidate) for candidate in rhs_candidates]:
-            union = list(set(lhs_candidate + rhs_candidate))
-            # Union should contains contiguous numbers from 0, otherwise it's same as another case.
-            if any(v >= len(union) for v in union):
-                continue
-            # Numbers in right but not in left should be sorted, otherwise it's same as another case.
-            only_in_right = [v for v in rhs_candidate if v not in lhs_candidate]
-            if any(only_in_right[i] > only_in_right[i + 1] for i in range(len(only_in_right) - 1)):
-                continue
-            combs = []
-            for i in range(1, len(union) + 1):
-                combs += list(itertools.combinations(union, i))
-            output_candidates = []
-            for comb in combs:
-                output_candidates += list(itertools.permutations(comb))
-            # When output_candidates is too many, it will take long time to run. Sample part of them.
-            random.shuffle(output_candidates)
-            output_candidates = output_candidates[:8]
-            for output_candidate in [list(candidate) for candidate in output_candidates]:
-                all_cases.append((lhs_candidate, rhs_candidate, output_candidate))
-
-    for case in all_cases:
-        equation = to_string(case[0]) + ',' + to_string(case[1]) + '->' + to_string(case[2])
-        pos1 = equation.find(',')
-        pos2 = equation.find('->')
-        lhs_op = equation[0:pos1]
-        rhs_op = equation[pos1 + 1:pos2]
-        lhs_shape = []
-        for c in lhs_op:
-            lhs_shape.append(SIZE_MAP[c.upper()])
-        rhs_shape = []
-        for c in rhs_op:
-            rhs_shape.append(SIZE_MAP[c.upper()])
-
-        pt_model = NeuralNetEinsum(lhs_shape[-1]).to(device)
-        ort_model = ORTModule(copy.deepcopy(pt_model))
-
-        def run_step(model, input_left, input_right):
-            prediction = model(input_left, input_right)
-            loss = prediction.sum()
-            loss.backward()
-            return prediction
-
-        for _ in range(5):
-            pt_input_left = torch.rand(lhs_shape, device=device)
-            pt_input_right = torch.rand(rhs_shape, device=device)
-            ort_input_left = copy.deepcopy(pt_input_left)
-            ort_input_right = copy.deepcopy(pt_input_right)
-            pt_prediction = run_step(pt_model, pt_input_left, pt_input_right)
-            ort_prediction = run_step(ort_model, ort_input_left, ort_input_right)
-
-            _test_helpers.assert_values_are_close(ort_prediction, pt_prediction, atol=1e-3, rtol=1e-3)
-            _test_helpers.assert_gradients_match_and_reset_gradient(ort_model, pt_model, atol=1e-3, rtol=1e-3)
+# def test_gradient_correctness_einsum_2():
```

Comment:
skipping it would be better than commenting it out.

Reply:
Right! Actually I can disable them for specific torch version only. Thanks!